### ISSUE TITLE
AssistantMK2 is not activated by Hotword 2.0

### ISSUE THREAD

**Issue by ivanovd** [eouia/MMM-Hotword#43]
Here is my config:

```
{
  module: "MMM-AssistantMk2",
  position: "top_right",
  config: {
    record: {
        recordProgram : "arecord",
        device        : "plughw:1",
    },
  notifications: {
  ASSISTANT_ACTIVATED: "HOTWORD_PAUSE",
  ASSISTANT_DEACTIVATED: "HOTWORD_RESUME",
},
  screenDuration: 1000,
  responseVoice: true,
  responseScreen: true,
  responseAlert: true,
  }
},

{
    module: "MMM-Hotword",
    position: "top_right",
    config: {
      mic: {
        recordProgram : "arecord",
        device        : "plughw:1",
      },
      models: [
        {
          hotwords: ["assistant"],
          file: "smart_mirror.umdl",
          sensitivity: "0.5",
        },
      ],
      commands: {
        "assistant": { //When `smart mirror` be spoken, MMM-AssistantMk2 will be activated.
          notificationExec: {
            notification: "ASSISTANT_ACTIVATE",
            payload: (hotword, file) => {
              return {profile: "default"}
            }
          },
          restart:false
        }
      }
    },
},
```

Evrything loads fine and now I see a AssistantMK2 icon and a Hotword icon. When I speak the hotword the word "assistant" shows next to the Hotword icon and stays there, however, AssistantMK2 is not activated.

![MaxthonSnap20190519200328](https://user-images.githubusercontent.com/4889583/57985864-e3515a80-7a76-11e9-8bcf-05bb8311d6a8.png)

The previous version was working fine.

What am I doing wrong?

Thanks

**Comment by ivanovd** [comment]
The previous version was working fine.

**Comment by eouia** [comment]
can you show me the logs?

**Comment by ivanovd** [comment]
````
tail -500 /home/pi/.pm2/logs/MagicMirror-out.log
[HOTWORD] After Recording: 8000
[HOTWORD] After Recording: 4000
[HOTWORD] After Recording: 4000
[HOTWORD] After Recording: 4000
[HOTWORD] After Recording: 4000
[HOTWORD] After Recording: 8000
[HOTWORD] After Recording: 8000
[HOTWORD] After Recording: 8000
[HOTWORD] After Recording: 4000
[HOTWORD] After Recording: 4000
[HOTWORD] After Recording: 8000
[HOTWORD] After Recording: 8000
[HOTWORD] After Recording: 8000
[HOTWORD] After Recording: 8000
[HOTWORD] After Recording: 8000
[HOTWORD] After Recording: 8000
[HOTWORD] After Recording: 8000

````

**Comment by eouia** [comment]
hmmm. i'll check tomorrow. sorry wait for a while.

**Comment by eouia** [comment]
It seems afterRecording couldn't stop. It is designed to stop automatically when there is no sound-input.
There are some possibilities;
1) Mic is too sensitive so white noise or any other sound(e.g: fan noise, air ventilation....) be caught.
2) Your environment is somewhat noisy.
=> Adjust your mic volume.

Or maybe my module has a bug. I'll check it. 
Anyway, usually ppl don't need afterRecord feature at this moment.(I made it for continuous recording to update next AssistantMk2 feature), so I'll patch for disabling afterRecord feature. Sorry, wait for a while.

**Comment by eouia** [comment]
I added `afterRecordLimit` to disable or stop by force. See the command.md part.

**Comment by Chanuka-Jayanath** [comment]
I've got the same problem, even when I mannually clicks on the Google buttons it launches but not listening ? Anyway the hotward has been detected on the screnn when i say "computer"

**Comment by eouia** [comment]
@Chanuka-Jayanath Give me the log. We need to confirm where that issue came from. And if you want to activate the touch, don't use Hotword together.
Anyway, I've updated 2 hours ago. try it first.

**Comment by Chanuka-Jayanath** [comment]
I've tried the updated version It just worked once and stops working, I don't use the touch i just did it with vnc. How can I take the Log ?

**Comment by eouia** [comment]
`pm2 logs` or `npm start dev`(before you should stop pm2)

**Comment by Chanuka-Jayanath** [comment]
[TAILING] Tailing last 15 lines for [all] processes (change the value with --lines option)
/home/pi/.pm2/pm2.log last 15 lines:
PM2        | 2019-05-20T19:59:17: PM2 log: pid=2510 msg=failed to kill - retrying in 100ms
PM2        | 2019-05-20T19:59:17: PM2 log: pid=2510 msg=failed to kill - retrying in 100ms
PM2        | 2019-05-20T19:59:17: PM2 log: pid=2510 msg=failed to kill - retrying in 100ms
PM2        | 2019-05-20T19:59:17: PM2 log: pid=2510 msg=failed to kill - retrying in 100ms
PM2        | 2019-05-20T19:59:17: PM2 log: pid=2510 msg=failed to kill - retrying in 100ms
PM2        | 2019-05-20T19:59:17: PM2 log: pid=2510 msg=failed to kill - retrying in 100ms
PM2        | 2019-05-20T19:59:17: PM2 log: pid=2510 msg=failed to kill - retrying in 100ms
PM2        | 2019-05-20T19:59:18: PM2 log: pid=2510 msg=failed to kill - retrying in 100ms
PM2        | 2019-05-20T19:59:18: PM2 log: pid=2510 msg=failed to kill - retrying in 100ms
PM2        | 2019-05-20T19:59:18: PM2 log: pid=2510 msg=failed to kill - retrying in 100ms
PM2        | 2019-05-20T19:59:18: PM2 log: pid=2510 msg=failed to kill - retrying in 100ms
PM2        | 2019-05-20T19:59:18: PM2 log: pid=2510 msg=failed to kill - retrying in 100ms
PM2        | 2019-05-20T19:59:18: PM2 log: Process with pid 2510 still alive after 1600ms, sending it SIGKILL now...
PM2        | 2019-05-20T19:59:18: PM2 log: App [mm:0] exited with code [0] via signal [SIGKILL]
PM2        | 2019-05-20T19:59:18: PM2 log: pid=2510 msg=process killed

/home/pi/.pm2/logs/mm-error.log last 15 lines:
0|mm       | (node:1652) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
0|mm       | 1
0|mm       | ATTENTION: default value of option force_s3tc_enable overridden by environment.
0|mm       | (node:1838) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
0|mm       | 1
0|mm       | ATTENTION: default value of option force_s3tc_enable overridden by environment.
0|mm       | ATTENTION: default value of option force_s3tc_enable overridden by environment.
0|mm       | (node:2076) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
0|mm       | 1
0|mm       | ATTENTION: default value of option force_s3tc_enable overridden by environment.
0|mm       | (node:2233) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
0|mm       | 1
0|mm       | ATTENTION: default value of option force_s3tc_enable overridden by environment.
0|mm       | (node:2386) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
0|mm       | 1

/home/pi/.pm2/logs/mm-out.log last 15 lines:
0|mm       | Connecting socket for: MMM-PIR-Sensor
0|mm       | Connecting socket for: MMM-Paris-RATP-PG
0|mm       | Connecting socket for: MMM-PushBulletNotifications
0|mm       | Connecting socket for: MMM-NowPlayingOnSpotify
0|mm       | Connecting socket for: MMM-AssistantMk2
0|mm       | Connecting socket for: calendar
0|mm       | Starting node helper for: calendar
0|mm       | Connecting socket for: newsfeed
0|mm       | Starting module: newsfeed
0|mm       | Connecting socket for: MMM-NotificationTrigger
0|mm       | Connecting socket for: MMM-Hotword
0|mm       | [HOTWORD] MMM-Hotword starts
0|mm       | Connecting socket for: MMM-network-signal
0|mm       | MMM-network-signal helper started ...
0|mm       | Sockets connected & modules started ...

**Comment by eouia** [comment]
To make things simple, disable all other modules except Hotword and AMK2, then show me the logs when the issues happen. I recommend to use `npm start dev` to get realtime log on your terminal

**Comment by Chanuka-Jayanath** [comment]
I just found the problem, as you said the mic sensibilty affects the module to work properly anyway for now it seems to be working fine if i will find any other issues i'll get back to you. Thank you very much you are doing such a great job ! good luck !

**Comment by ivanovd** [comment]
I lowered the mic sensitivity and updated to Hotword ver. 2.0.1. It was had to get the correct value since it is a far-field 2 mic array and it is very sensitive as you can imagine. It works now though :) Thanks.

P.S. Is there a way to remove the icon from showing?

**Comment by Gandohr** [comment]
![F652572B-B617-4F0C-8E20-CFFF36F2503C](https://user-images.githubusercontent.com/218680/58173045-91891a00-7c9a-11e9-85f7-c5b97ad1b61e.jpeg)

I use the same config settings. This is what mine doe.
The hotword icon on screen shows the hotword and Mk2 is activated, both then just wait on input but do not respond the anything

**Comment by Gandohr** [comment]
> I lowered the mic sensitivity and updated to Hotword ver. 2.0.1. It was had to get the correct value since it is a far-field 2 mic array and it is very sensitive as you can imagine. It works now though :) Thanks.
> 
> P.S. Is there a way to remove the icon from showing?

Remove the position line from the MMM-Hotword config to top it from showing

**Comment by eouia** [comment]
@ivanovd 
>P.S. Is there a way to remove the icon from showing?

Use `useDisplay:false`


@Gandohr 
Speaker error usually comes from invalid ALSA configuration.(or H/W, S/W troubles)
The reason would be various.

Anyway, that seems not any more being related with Hotword module itself. I think new issue topic is better to focus new trouble.

**Comment by Gandohr** [comment]
> @ivanovd
> 
> > P.S. Is there a way to remove the icon from showing?
> 
> Use `useDisplay:false`
> 
> @Gandohr
> Speaker error usually comes from invalid ALSA configuration.(or H/W, S/W troubles)
> The reason would be various.
> 
> Anyway, that seems not any more being related with Hotword module itself. I think new issue topic is better to focus new trouble.

Thanks, but it always gave the speaker error but always worked.
Don’t thing the problem is the speaker error but the fact hotword stays open or something.
System always worked, only thing that changed is hotword, seems unlogical to be alsa config as that did not change, and the mic and sound tests work.

**Comment by eouia** [comment]
@Gandohr 
Try `chimeOnFinish:null` of MMM-Hotword configuration. I think this error is caused from device occupation. ALSA of RPI is not so smart, so by default, it cannot play two sound-streams together unless proper mixer configuration is prepared. In this case, the chime (after detection) is played by MMM-Hotword, and almost simultaneously MMM-AssistantMk2 tried it's own chime. but speaker was occupied already by MMM-Hotword, that error would happen.
In my previous MMM-Hotword, it didn't use any sound output, so there was no problem about this.
To avoid this, the best solution is to configure your Audio ALSA setup to mix multiple streams. Or using PulseAudio to mix multi-source...
Or Allow Only AssistantMk2 to use speaker. (`chimeOnFinish:null`)

**Comment by Gandohr** [comment]
![45AA94E2-BC8D-4A70-B3CD-5C22D435CE18](https://user-images.githubusercontent.com/218680/58193945-81863000-7cc4-11e9-879e-af7c858da042.jpeg)
I get this error now, but it did the trick it now works again.
Thanks

**Comment by eouia** [comment]
I'll close this issue. @ivanovd  and @Gandohr, you can make new issue topic to focus. I wish you always have a good luck.
